src/Vault.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: BSD Clause-3
+pragma solidity ^0.8.24;
+
+import {
+    AccessControlUpgradeable,
+    Address,
+    ERC20PermitUpgradeable,
+    IERC20,
+    Math,
+    ReentrancyGuardUpgradeable,
+    SafeERC20
+} from "./Common.sol";
+
+import {Storage} from "src/Storage.sol";
+import {IVault} from "src/interface/IVault.sol";
+
+contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
What's the intention on how to handle 

ynETH and ynLSDe existing yieldnest strategies?

Can they simply be classifed as deposit assets for the vault so that current existing ynETH and ynLSDe be directly deposited to ynETHx?

Reply:
These will be assets and strategies. Will pick up this question in the scenario tests for ynETH and ynLSDe